### PR TITLE
Fix Zuul's RequestContext.getOriginResponseHeaders not being populated

### LINKED ISSUE #2979: Zuul's RequestContext.getOriginResponseHeaders is no longer populated with the response headers
## Context

Zuul's  `com.netflix.zuul.context.RequestContext` exposes a method:

```java
// the Origin response headers
List<Pair<String, String>> getOriginResponseHeaders()
```

Which returns the list of http headers as returned by the downstream app behind the Zuul filter chain.

## Problem

This worked as expected in spring-cloud-netflix up to 1.3.x versions, but the behaviour changed starting from 1.4.0.

**Now this list contains one single header**: `X-Zuul-Service` and is no longer populated with the original response headers.

## Origin

I tracked down this regression to this pull request: https://github.com/spring-cloud/spring-cloud-netflix/pull/1965

The line which populates this field was *accidentally* (that's my theory at least) deleted in the midst of a refactoring:

screen shot 2018-05-28 at 15 55 43

A PR with the (trivial fix) will follow shortly, with an anti-regression test.

### PR DESCRIPTION
**Pull request by jawher**
This PR proposes a fix the the regression described in this issue: #2979

tl;dr: RequestContext.getOriginResponseHeaders was no longer populated
with the original responses headers due to a regression caused by a
refactor.

An anti-regression test was also added.

Fixes #2979

### CODE REVIEW

**Comment by jawher** [comment]
Not sure how to proceed in order to have this fix backported to the 1.4.x branch.

Should I create a second pull request targeting the 1.4 branch or will the maintainers just cherry-pick the merge commit ?
